### PR TITLE
Chore: Remove methods from sqlstore

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:
This Pr removes some methods from user.go of sqlstore. The enterprise fix for this PR: https://github.com/grafana/grafana-enterprise/pull/4044.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -260,6 +260,7 @@ require (
 	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.6.3
 	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.6.3
 	gocloud.dev v0.25.0
+	gotest.tools v2.2.0+incompatible
```
Comment (sakjur) [inline go.mod:263]: 🤔 Do we use this? I don't see where 🙈

Comment (idafurjes) [reply go.mod:263]: I had to run go mid tidy, because of enterprise, I am not sure why

Comment (idafurjes) [reply go.mod:263]: https://github.com/grafana/grafana-enterprise/pull/4044/files

File: pkg/cmd/grafana-cli/commands/conflict_user_command_test.go [hunk 2]
```diff
@@ -629,11 +628,6 @@ func TestMergeUser(t *testing.T) {
 			// test starts here
 			err = r.MergeConflictingUsers(context.Background())
 			require.NoError(t, err)
-
-			// user with uppercaseemail should not exist
-			query := &models.GetUserByIdQuery{Id: userWithUpperCase.ID}
-			err = sqlStore.GetUserById(context.Background(), query)
-			require.Error(t, user.ErrUserNotFound, err)
```
Comment (zserge) [inline pkg/cmd/grafana-cli/commands/conflict_user_command_test.go:636]: I think this was the essence of the test, so either comment it out/skip completely or maybe bring in the user.Service to check for user's existence.

Comment (idafurjes) [reply pkg/cmd/grafana-cli/commands/conflict_user_command_test.go:636]: I reached out to the author of the test, the solution would be using fakes as intended

File: go.mod (hunk 1 above)
Comment (sakjur) [reply go.mod:263]: I looked it up, it looks like it's an indirect dependency 🤷 I'm not sure why it doesn't show up as such, but eh :+1:

Comment (idafurjes) [reply go.mod:263]: interesting 🤷‍♀️

**Review by sakjur — APPROVED** [review]
LGTM and better merged than not, but it'd be nice to follow up on what we'd need to make `TestMergeUser` a more useful test again 🙂

**Comment by idafurjes** [comment]
there will be a follow up issue, @eleijonmarck was kind to agree to fix the test <3

**Pull request merged**